lib/_http_parser.js

```diff
@@ -0,0 +1,907 @@
+                  break;
+              }
+              if (i === 10) {
+                if (fieldValue.search(RE_CONN_CLOSE) > -1)
+                  this._flags |= FLAG_CONNECTION_CLOSE;
+                if (fieldValue.search(RE_CONN_KEEPALIVE) > -1)
+                  this._flags |= FLAG_CONNECTION_KEEP_ALIVE;
+                if (fieldValue.search(RE_CONN_UPGRADE) > -1)
+                  this._flags |= FLAG_CONNECTION_UPGRADE;
+              }
+              break;
+            case 14:
+              // Somewhere between 10 and 14 characters, the previously used
+              // for loop algorithm becomes slower than doing a search using a
+              // simple, case-insensitive, non-anchored regexp
+              if (fieldValue.length > 0 &&
+                  fieldName.search(RE_CONTLEN) === 0) {
+                var val = parseInt(fieldValue, 10);
+                if (val !== val || val > MAX_CHUNK_SIZE)
+                  return this._setError('Bad Content-Length');
+                this._contentLen = val;
+              }
+              break;
+            case 17:
+              if (fieldName.search(RE_TE) === 0 &&
+                  fieldValue.search(RE_TE_CHUNKED) > -1) {
+                this._flags |= FLAG_CHUNKED;
+              }
+              break;
+          }
+          break;
+        case STATE_REQ_LINE:
+          // Original HTTP parser ignored blank lines before request/status
+          // line, so we do that here too ...
+          if (buflen === 0 || buf.charCodeAt(0) === CR)
+            break;
+
+          var firstSP;
+          var urlStart;
+          var urlEnd;
+          var minor;
+          var end = (buf.charCodeAt(buflen - 1) === CR ?
+                     buflen - 3 : buflen - 2);
+          // Start working backwards and both validate that the line ends in
+          // ` HTTP/1.[01]` and find the end of the URL (in case there are
+          // multiple spaces/tabs separating the URL and HTTP version
+          var ch = buf.charCodeAt(end + 1);
+          if (ch === 49)
+            minor = 1;
+          else if (ch === 48)
+            minor = 0;
+          else
+            return this._setError('Malformed request line');
+          var h = 0;
+          while (end >= 0) {
+            var ch = buf.charCodeAt(end);
+            if (h < 8) {
+              if (ch !== REQ_HTTP_VER_BYTES[h++])
+                return this._setError('Malformed request line');
+            } else if (ch >= 33 && ch !== 127) {
+              urlEnd = end + 1;
+              break;
+            }
+            --end;
+          }
+          if (urlEnd === undefined)
+            return this._setError('Malformed request line');
+
+          // Now start working forwards and both validate the HTTP method and
+          // find the start of the URL (in case there are multiple spaces/tabs
+          // separating the method and the URL
+          var isConnect = false;
+          var c = 0;
```

Comment:
`c` is a very ambiguous variable name here. Perhaps you could change it to `connectChars` or something along that line? The code (below, dealing with `CONNECT`) isn't immediately obvious on what its doing either, and a comment would help.